presto-accumulo/src/main/java/com/facebook/presto/accumulo/index/IndexLookup.java

```diff
@@ -0,0 +1,425 @@
+            scan.fetchColumnFamily(cf);
+
+            // For each entry in the scanner
+            Set<Range> columnRanges = new HashSet<>();
+            for (Entry<Key, Value> entry : scan) {
+                entry.getKey().getColumnQualifier(tmpCQ);
+
+                // Add to our column ranges if it is in one of the row ID ranges
+                if (inRange(tmpCQ, rowIDRanges)) {
+                    columnRanges.add(new Range(tmpCQ));
+                }
+            }
+
+            LOG.info("Retrieved %d ranges for column %s", columnRanges.size(),
+                    e.getKey().getName());
+
+            // If finalRanges is null, we have not yet added any column ranges
+            if (finalRanges == null) {
+                finalRanges = new HashSet<>();
+                finalRanges.addAll(columnRanges);
+            }
+            else {
+                // Retain only the row IDs for this column that have already been added
+                // This is your set intersection operation!
+                finalRanges.retainAll(columnRanges);
+            }
+
+            // Close the scanner
+            scan.close();
+        }
+
+        // Return the final ranges for all constraint pairs
+        return new ArrayList<>(finalRanges);
+    }
+
+    /**
+     * Bins the given collection of ranges into metadata for the Presto splits
+     *
+     * @param numRangesPerBin Number of ranges to put in a single metadata object
+     * @param splitRanges The ranges to add
+     * @param prestoSplits The output collection for the tablet metadata
+     */
+    private void binRanges(int numRangesPerBin, List<Range> splitRanges,
+            List<TabletSplitMetadata> prestoSplits)
+    {
+        // Shuffle the handles to give an even distribution across all tablet splits
+        Collections.shuffle(splitRanges);
+
+        // Location here doesn't matter, I think this 9997 is left over from the connector example
+        // Let's leave it to pay homage to where this thing came from
+        String loc = "localhost:9997";
+        int toAdd = splitRanges.size();
+        int fromIndex = 0;
+        int toIndex = Math.min(toAdd, numRangesPerBin);
+        do {
+            // Add the sublist of range handles
+            prestoSplits.add(new TabletSplitMetadata(loc, splitRanges.subList(fromIndex, toIndex)));
+            toAdd -= toIndex - fromIndex;
+            fromIndex = toIndex;
+            toIndex += Math.min(toAdd, numRangesPerBin);
+        }
+        while (toAdd > 0);
+    }
+
+    /**
+     * Gets a Boolean value indicating if the given value is in one of the Ranges in the given
+     * collection
+     *
+     * @param t Text object to check against the Range collection
+     * @param ranges Ranges to look into
+     * @return True if the text object is in one of the ranges, false otherwise
+     */
+    private boolean inRange(Text t, Collection<Range> ranges)
+    {
```

Comment:
Maybe `ranges.stream().anyMatch(...)` instead? This gives the benefit of easier parallelization.